### PR TITLE
Unit test for mount/unmount snapshot and other features

### PR DESCRIPTION
**Pull request by imran-ansari**
*Implemented unit tests for mount/unmount snapshot
*Made changes for virtualCopyOf key
*New mock data added
*Get volume tester needed changes

@hpe-storage , @wdurairaj , @nilangekarss  - Please review the changes

NOTE: One test case for each protocol will remain in broken state till some more check-ins

### CODE REVIEW

File: hpedockerplugin/volume_manager.py [hunk 1]
```diff
@@ -218,10 +209,18 @@ def create_snapshot(self, src_vol_name, snapshot_name,
         snap_qos = volume.DEFAULT_QOS
         mount_cononflict_delay = vol['mount_conflict_delay']
 
+        is_snap = True
         snap_vol = volume.createvol(snapshot_name, snap_size, snap_prov,
                                     snap_flash, snap_compression, snap_qos,
                                     mount_cononflict_delay, is_snap)
 
+        # TODO: Remove the below code or remove snapshot back reference entry
```
Comment (nilangekarss) [inline hpedockerplugin/volume_manager.py]: @imran-ansari line #217 to #233 can be removed

File: test/fake_3par_data.py [hunk 2]
```diff
@@ -134,28 +139,119 @@
...
+    'retention_hours': '10'
+}
+
+snap1 = {
+    'name': SNAPSHOT_NAME1,
+    'id': SNAPSHOT_ID1,
+    'display_name': SNAPSHOT_NAME1,
+    'parent_id': VOLUME_ID,
+    'ParentName': VOLUME_NAME,
+    'expiration_hours': '10',
```
Comment (nilangekarss) [inline test/fake_3par_data.py]: expiration_hours and retention_hours are part of snap_metadata and not of snapshot dictionary

File: test/fake_3par_data.py [hunk 3]
```diff
@@ -134,28 +139,119 @@
...
+    'retention_hours': '10'
+}
+
+snap2 = {
+    'name': SNAPSHOT_NAME2,
+    'id': SNAPSHOT_ID2,
+    'display_name': SNAPSHOT_NAME2,
+    'parent_id': VOLUME_ID,
+    'ParentName': VOLUME_NAME,
+    'expiration_hours': '5',
```
Comment (nilangekarss) [inline test/fake_3par_data.py]: expiration_hours and retention_hours are part of snap_metadata and not of snapshot dictionary

File: test/getvolume_tester.py [hunk 4]
```diff
@@ -127,3 +88,66 @@ def check_response(self, resp):
         # in the flow or not
         mock_3parclient = self.mock_objects['mock_3parclient']
         mock_3parclient.getWsApiVersion.assert_called()
+
+
+class GetSnapshotUnitTest(GetVolumeUnitTest):
```
Comment (nilangekarss) [inline test/getvolume_tester.py]: I don't see implementation for this class

Comment (imran-ansari) [reply test/getvolume_tester.py]: It's a base class for snapshot-testers like TestSyncSnapshots

File: test/fake_3par_data.py (hunk 3 above)
Comment (imran-ansari) [reply test/fake_3par_data.py]: ok

File: test/fake_3par_data.py (hunk 2 above)
Comment (imran-ansari) [reply test/fake_3par_data.py]: ok

File: hpedockerplugin/volume_manager.py (hunk 1 above)
Comment (imran-ansari) [reply hpedockerplugin/volume_manager.py]: Sure

**Pull request merged**